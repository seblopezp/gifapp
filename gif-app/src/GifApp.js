import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifList } from "./components/GifList";

export const GifApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <div>
      <h1 className="title">Gif App</h1>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => (
          <GifList key={category} category={category} />
        ))}
      </ol>
    </div>
  );
};
