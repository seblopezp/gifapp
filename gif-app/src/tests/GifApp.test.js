import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { GifApp } from "../GifApp";
const wrapper = shallow(<GifApp />);
describe("Pruebas en <GifApp />", () => {
  test("Debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar lista de categorias", () => {
    const categories = ["Studio Ghibli", "Jujutsu Kaisen"];
    const wrapper = shallow(<GifApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifList").length).toBe(categories.length);
  });
});
