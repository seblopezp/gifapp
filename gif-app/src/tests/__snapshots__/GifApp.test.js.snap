// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifApp /> Debe mostrar el componente correctamente 1`] = `
<div>
  <h1
    className="title"
  >
    Gif App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`Pruebas en <GifApp /> Debe mostrar lista de categorias 1`] = `
<div>
  <h1
    className="title"
  >
    Gif App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifList
      category="Studio Ghibli"
      key="Studio Ghibli"
    />
    <GifList
      category="Jujutsu Kaisen"
      key="Jujutsu Kaisen"
    />
  </ol>
</div>
`;
